catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/transformer/TransformerProducer.java

```diff
@@ -87,34 +86,44 @@ public void process(Exchange exchange)
     if (mimeType != null) {
       in.removeHeader(CatalogComponent.MIME_TYPE_PARAMETER);
     } else {
-      LOGGER.debug("MIME Type retrieved from CamelCatalogEndpoint");
+      LOGGER.debug(
+          "No mimeType provided, defaulting to Camel CatalogEndpoint mimeType of [{}]",
+          endpoint.getMimeType());
       mimeType = endpoint.getMimeType();
     }
 
     LOGGER.debug("MIME Type = {}", mimeType);
 
     MimeTypeToTransformerMapper mapper = endpoint.getComponent().getMimeTypeToTransformerMapper();
 
+    Object metacard;
     if (mapper != null) {
       LOGGER.debug("Got a MimeTypeToTransformerMapper service");
 
-      metacard = transform(in, metacard, mimeType, transformerId, mapper);
+      metacard = transform(in, mimeType, transformerId, mapper);
     } else {
       LOGGER.debug("Did not find a MimeTypeToTransformerMapper service");
       throw new CatalogTransformerException("Did not find a MimeTypeToTransformerMapper service");
     }
 
-    // Set the response output to the Metacard from the transformation
-    exchange.getOut().setBody(metacard);
+    // Set the body to the Metacard from the transformation
+    in.setBody(metacard);
 
-    LOGGER.debug("EXITING: process");
+    LOGGER.trace("EXITING: process");
+  }
+
+  protected String getHeaderAsStringAndRemove(Message message, String key) {
```

Comment:
Can this be `private`?

Reply:
Moved to `InputTransformProducer` and made private.